### PR TITLE
shared/web/respond: introduce Responder and Response

### PR DESCRIPTION
**Pull request by amery**
a Responder is a narrowed view of the Registry only containing the types supported by a particular resource. when combined with a *http.Request it creates a Response with the correct renderer and provides shortcuts for responding

### CODE REVIEW

File: shared/web/respond/redirect.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	_, err = fmt.Fprintln(rw, http.StatusText(code), "Redirecting to", loc)
+	return err
+}
+
+// MovedPermanently sends a 301 Moved Permanently Response
+func (r *Response) MovedPermanently(rw http.ResponseWriter, loc string) error {
+	return r.redirect(rw, loc, http.StatusMovedPermanently)
+}
+
+// Found sends a 302 Moved Permanently Response
```
Comment (karasz) [inline shared/web/respond/redirect.go:40]: The comment is wrong, 302 is Found. The code is right, the comment is not.

**Review by karasz — CHANGES_REQUESTED** [review]

File: shared/web/respond/redirect.go (hunk 1 above)
Comment (amery) [reply shared/web/respond/redirect.go]: oops, amended.

**Review by karasz — APPROVED** [review]
LGTM now

**Comment by amery** [comment]
`shared/web/v0.3.3` tagged

**Pull request merged**